ext/drwrap/drwrap.c

```diff
@@ -2592,7 +2592,7 @@ drwrap_event_restore_state_ex(void *drcontext, bool restore_memory,
                 scan_stop = (app_pc *)pt->app_esp[i + 1];
```

Comment:
You mentioned offline that you also wanted to add a test?

Reply:
That was before I found that drwrap-test-detach reproduced the hang, and is already targeting the return address replacement, but was just disabled on arm.